tests/resources/test_predictor.py

```diff
@@ -278,30 +278,70 @@ def test_train(valid_graph_predictor_data):
     assert session.calls == expected_calls
 
 
-def test_list_predictors(valid_graph_predictor_data, valid_graph_predictor_data_empty):
+def test_list(valid_graph_predictor_data, valid_graph_predictor_data_empty):
     # Given
     session = FakeSession()
     collection = PredictorCollection(uuid.uuid4(), session)
     session.set_responses(
         {
             'response': [valid_graph_predictor_data, valid_graph_predictor_data_empty],
-            'next': ''
+            'page': 1,
+            'per_page': 25
         },
         basic_predictor_report_data,
         basic_predictor_report_data
     )
 
     # When
-    predictors = list(collection.list(per_page=20))
+    predictors = list(collection.list(per_page=25))
 
     # Then
-    expected_call = FakeCall(method='GET', path='/projects/{}/predictors'.format(collection.project_id),
-                                   params={'per_page': 20, 'page': 1})
+    expected_call = FakeCall(method='GET',
+                             path='/projects/{}/predictors'.format(collection.project_id),
+                             params={'per_page': 25, 'page': 1, 'archived': False})
     assert 1 == session.num_calls, session.calls
     assert expected_call == session.calls[0]
     assert len(predictors) == 2
 
 
+def test_list_all(valid_graph_predictor_data, valid_graph_predictor_data_empty):
+    # Given
+    session = FakeSession()
+    collection = PredictorCollection(uuid.uuid4(), session)
+    session.set_responses(
+        {'response': [valid_graph_predictor_data, valid_graph_predictor_data_empty]},
+        basic_predictor_report_data,
+        basic_predictor_report_data
+    )
+
+    # When
+    predictors = list(collection.list_all(per_page=25))
+
+    # Then
+    expected_call = FakeCall(method='GET',
+                             path='/projects/{}/predictors'.format(collection.project_id),
+                             params={'per_page': 25, 'page': 1})
+    assert 1 == session.num_calls, session.calls
+    assert expected_call == session.calls[0]
+    assert len(predictors) == 2
+
+
+def test_list_archived(valid_graph_predictor_data):
+    # Given
```

Comment:
this structure is nice

Reply:
Agreed. I try to follow something of the sort, even if I don't put the comments in there to clearly denote it.